Neos.Flow/Classes/Persistence/Doctrine/AllowedObjectsListener.php

```diff
@@ -0,0 +1,118 @@
+<?php
+namespace Neos\Flow\Persistence\Doctrine;
+
+/*
+ * This file is part of the Neos.Flow package.
+ *
+ * (c) Contributors of the Neos Project - www.neos.io
+ *
+ * This package is Open Source Software. For the full copyright and license
+ * information, please view the LICENSE file which was distributed with this
+ * source code.
+ */
+
+use Doctrine\DBAL\Exception\ConnectionException;
+use Doctrine\ORM\Event\OnFlushEventArgs;
+use Neos\Flow\Annotations as Flow;
+use Neos\Flow\Log\ThrowableStorageInterface;
+use Neos\Flow\Log\Utility\LogEnvironment;
+use Neos\Flow\Persistence\AllowedObjectsContainer;
+use Neos\Flow\Persistence\Exception as PersistenceException;
+use Neos\Flow\Persistence\PersistenceManagerInterface;
+use Psr\Log\LoggerInterface;
+
+/**
+ * An onFlush listener for Flow's Doctrine PersistenceManager, that validates to be persisted entities
+ * against the list of allowed objects.
+ *
+ * This listener is outsourced from the PersistenceManager to avoid recursive dependencies when building
+ * the EntityManager.
+ *
+ * @Flow\Scope("singleton")
+ * @api
+ */
+class AllowedObjectsListener
+{
+    /**
+     * @Flow\Inject
+     * @var LoggerInterface
+     */
+    protected $logger;
+
+    /**
+     * @Flow\Inject
+     * @var ThrowableStorageInterface
+     */
+    protected $throwableStorage;
+
+    /**
+     * @Flow\Inject
+     * @var AllowedObjectsContainer
+     */
+    protected $allowedObjects;
+
+    /**
+     * @Flow\Inject(lazy=true)
+     * @var PersistenceManagerInterface
+     */
+    protected $persistenceManager;
+
+    /**
+     * Doctrine onFlush listener that checks for only allowed objects and reconnects
+     * if the database connection was closed.
+     *
+     * @param OnFlushEventArgs $args
+     * @throws PersistenceException
+     */
+    public function onFlush(OnFlushEventArgs $args)
```

Comment:
1:1 copy from the PersistenceManager apart from the `$this->allowedObjects->shouldCheck()` check